apps/patient/src/components/PharmacyInfo.tsx

```diff
@@ -16,41 +28,154 @@ interface HoursProps {
   is24Hr?: boolean;
   opens?: string;
   closes?: string;
+  hours?: EnrichedPharmacy['hours'];
+  showHours?: boolean;
 }
 
-const Hours = ({ is24Hr, isOpen, isClosingSoon, opens, closes }: HoursProps) => {
-  const color = isClosingSoon ? 'orange.500' : isOpen ? 'green' : 'red';
-  const text = is24Hr ? t.open24hrs : isClosingSoon ? t.closingSoon : isOpen ? t.open : t.closed;
-  const hasHours = isOpen != null;
+const daysOfWeek = ['SUNDAY', 'MONDAY', 'TUESDAY', 'WEDNESDAY', 'THURSDAY', 'FRIDAY', 'SATURDAY'];
 
-  if (!hasHours) return null;
+const hoursLookup = Object.fromEntries(daysOfWeek.map((d, i) => [d, i]));
 
-  return (
-    <HStack w="full" whiteSpace="nowrap" overflow="hidden" height="fit-content">
-      {isOpen != null || isClosingSoon ? (
-        <Text fontSize="sm" color={color}>
-          {text}
+const formatTime = (hstr: string) => {
+  const [h, m] = hstr.split(':');
+  const hoursInt = parseInt(h, 10);
+  return `${hoursInt === 12 ? 12 : hoursInt % 12}:${m} ${hoursInt >= 12 ? 'pm' : 'am'}`;
+};
+
+const HoursRow = ({
+  hours,
+  is24Hr,
+  isClosed,
+  dayOfWeek
+}: {
+  hours?: Array<{
+    openFrom: string;
+    openUntil: string;
+  }>;
+  is24Hr: boolean;
+  isClosed: boolean;
+  dayOfWeek: string;
+}) => {
+  if (isClosed || is24Hr) {
+    return (
+      <HStack w="full" justifyContent={'space-between'}>
+        <Text fontSize="sm" color="gray.500">
+          {titleCase(dayOfWeek)}
         </Text>
-      ) : null}
-      {!is24Hr && ((isOpen && closes) || (!isOpen && opens)) ? (
-        <Text color="gray.500">&bull;</Text>
-      ) : null}
-      {!is24Hr && isClosingSoon ? (
-        <Text fontSize="sm" color="gray.500" isTruncated>
-          {closes}
+        <Text fontSize="sm" color="gray.500">
+          {is24Hr ? 'Open 24hr' : 'Closed'}
         </Text>
-      ) : null}
-      {!is24Hr && !isClosingSoon && isOpen && closes ? (
-        <Text fontSize="sm" color="gray.500" isTruncated>
-          {closes}
+      </HStack>
+    );
+  }
+  if (hours == null || hours.length === 0) return null;
+  return (
+    <>
+      <HStack w="full" justifyContent={'space-between'}>
+        <Text fontSize="sm" color="gray.500">
+          {titleCase(dayOfWeek)}
         </Text>
-      ) : null}
-      {!is24Hr && !isClosingSoon && !isOpen && opens ? (
-        <Text fontSize="sm" color="gray.500" isTruncated>
-          {opens}
+        <Text fontSize="sm" color="gray.500">
+          {formatTime(hours[0].openFrom)} - {formatTime(hours[0].openUntil)}
         </Text>
-      ) : null}
-    </HStack>
+      </HStack>
+      {/* Include the rest of the hours but without day of week information */}
+      {hours.slice(1).map((h) => (
+        <HStack w="full" justifyContent={'space-between'} key={h.openFrom}>
+          <Text fontSize="sm" color="gray.500">
+            {''}
+          </Text>
```

Comment:
We need a value there, but we dont want to show the name of the day.

i.e. Do

```
SUNDAY 9-1
               2-7
```
rather than
```
SUNDAY 9-1
SUNDAY 2-7
```

Reply:
oh try <Spacer />